src/business_logic/transaction/objects/internal_declare.rs

```diff
@@ -0,0 +1,428 @@
+use std::collections::HashMap;
+
+use felt::{felt_str, Felt};
+use num_traits::Zero;
+
+use crate::{
+    business_logic::{
+        execution::{
+            error::ExecutionError,
+            execution_entry_point::ExecutionEntryPoint,
+            objects::{CallInfo, TransactionExecutionContext, TransactionExecutionInfo},
+        },
+        fact_state::state::ExecutionResourcesManager,
+        state::state_api::{State, StateReader},
+        transaction::{
+            error::TransactionError,
+            fee::{calculate_tx_fee, execute_fee_transfer, FeeInfo},
+        },
+    },
+    core::{
+        contract_address::starknet_contract_address::compute_class_hash,
+        transaction_hash::starknet_transaction_hash::calculate_declare_transaction_hash,
+    },
+    definitions::{general_config::StarknetGeneralConfig, transaction_type::TransactionType},
+    services::api::contract_class::{ContractClass, EntryPointType},
+    utils::{calculate_tx_resources, felt_to_hash, verify_no_calls_to_other_contracts, Address},
+};
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+///  Represents an internal transaction in the StarkNet network that is a declaration of a Cairo
+///  contract class.
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+pub(crate) struct InternalDeclare {
+    pub(crate) class_hash: [u8; 32],
+    pub(crate) sender_address: Address,
+    pub(crate) tx_type: TransactionType,
+    pub(crate) validate_entry_point_selector: Felt,
+    pub(crate) version: u64,
+    pub(crate) max_fee: u64,
+    pub(crate) signature: Vec<Felt>,
+    pub(crate) nonce: Felt,
+    pub(crate) hash_value: Felt,
+}
+
+// ------------------------------------------------------------
+//                        Functions
+// ------------------------------------------------------------
+impl InternalDeclare {
+    #![allow(unused)] // TODO: delete once used
+    pub fn new(
+        contract_class: ContractClass,
+        chain_id: Felt,
+        sender_address: Address,
+        max_fee: u64,
+        version: u64,
+        signature: Vec<Felt>,
+        nonce: Felt,
+    ) -> Result<Self, TransactionError> {
+        let hash = compute_class_hash(&contract_class)?;
+        let class_hash = felt_to_hash(&hash);
+
+        let hash_value = calculate_declare_transaction_hash(
+            contract_class,
+            chain_id,
+            sender_address.clone(),
+            max_fee,
+            version,
+            nonce.clone(),
+        )?;
+        // Value generated from get_selector_from_name(VALIDATE_DECLARE_ENTRY_POINT_NAME)
+        let validate_entry_point_selector = felt_str!(
+            "1148189391774113786911959041662034419554430000171893651982484995704491697075"
+        );
```

Comment:
Maybe just using the constant `VALIDATE_DEPLOY_ENTRY_POINT_SELECTOR` would be better?

Reply:
Is not the same selector, i could add it to the lazy_static structure.